File: 시나공/006 반복문/예제2.c
#include <stdio.h>
int main() {
    int a=0, hap=0;
    while(1) // 무한반복
    {
        a++;
        if(a>5)
            break; // 반복문 탈출. 바깥인 printf문으로
        if(a%2==0)
            continue; // 반복문 유지. 반복문의 처음인 a++로
        hap += a;
    }
    printf("%d, %d\n", a, hap);
}

// a=1 a%2==1 hap=1
// a=2 a%2==0
// a=3 a%2==1 hap=1+3=4
// a=4 a%2==0
// a=5 a%2==1 hap=4+5=9
// a=6

// 결과 6, 9

File: 시나공/012 Python의 활용/예제2.py
""" for문 """

a = [ 35, 55, 65, 84, 45 ]
hap = 0
for i in a: # a의 크기만큼 a의 값을 i에 저장하면서 반복
    hap += i
avg = hap / len(a)
print(hap, avg)

# i=a[0]=35 hap=35
# i=a[1]=55 hap=35+55
# i=a[2]=65 hap=35+55+65
# i=a[3]=84 hap=35+55+65+84
# i=a[4]=45 hap=35+55+65+84+45=284
# avg = hap/len(a) = 284/5 = 56.8

# 결과 284, 56.8